Week-6/maps&set/script.js

```diff
@@ -0,0 +1,45 @@
+// WeakMap for storing product views
+const productViews = new WeakMap();
+
+// WeakSet for storing product IDs that are in the user's cart
+const cartItems = new WeakSet();
+
+// Function to increment product views
+function incrementProductViews(productId) {
```

Comment:
In the current implementation, productRef is a new object created every time `incrementProductViews `or `addToCart `is called, which means it won't match the previous references stored in `productViews `or `cartItems`. Instead, these should be persistent references that correspond to product IDs.

Sharing for learning!